spark/src/main/scala/org/apache/comet/shims/ShimCometSparkSessionExtensions.scala

```diff
@@ -32,4 +33,9 @@ trait ShimCometSparkSessionExtensions {
     .map { a => a.setAccessible(true); a }
     .flatMap(_.get(scan).asInstanceOf[Option[Aggregation]])
     .headOption
+
+  def getOffset(limit: LimitExec): Option[Int] = limit.getClass.getDeclaredFields
```

Comment:
nit: I wonder if we can just return `0` if there is no offset in this method, so that we don't have to do `getOffset(op).getOrElse(0)` in a few places.